src/internal/pfsdb/branches.go

```diff
@@ -211,6 +223,12 @@ func DeleteBranch(ctx context.Context, tx *pachsql.Tx, id BranchID) error {
 	if _, err := tx.ExecContext(ctx, `DELETE FROM pfs.branches WHERE id = $1`, id); err != nil {
 		return errors.Wrapf(err, "could not delete branch %d", id)
 	}
+	if _, err := tx.ExecContext(ctx, `DELETE FROM pfs.branch_provenance WHERE from_id = $1 OR to_id = $1`, id); err != nil {
```

Comment:
do these statements need to come before we delete the branch since they point to an entry in the branches table?

Reply:
Good catch!